lib/Infer/InstSynthesis.cpp

```diff
@@ -1244,6 +1293,16 @@ bool InstSynthesis::isWiringInvalid(const LocVar &Left, const LocVar &Right) {
           InvalidWirings.count(std::make_pair(Right, Left)));
 }
 
+bool InstSynthesis::isInputConst(const LocVar &Loc) {
+  if (Loc.first != 0)
+    return false;
+
+  if (!LocInstMap.count(getLocVarStr(Loc, CONST_PREFIX)))
```

Comment:
I think this statement would be better written as "return LocInstMap.count(getLocVarStr(Loc, CONST_PREFIX) != 0;"

Reply:
Sure. The statement is ugly anyway, the data structures in synthesis "evolved" over time and are far away from being well-designed.